models/src/main/scala/coop/rchain/models/FringeData.scala

```diff
@@ -0,0 +1,64 @@
+package coop.rchain.models
+
+import com.google.protobuf.ByteString
+import coop.rchain.casper.protocol.FringeDataProto
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.syntax._
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import scodec.bits.ByteVector
+
+final case class FringeData(
+    fringeHash: Blake2b256Hash,
+    fringe: Set[BlockHash],
+    fringeDiff: Set[BlockHash],
+    stateHash: Blake2b256Hash,
+    // Rejected data in finalized fringe
+    rejectedDeploys: Set[ByteString],
+    rejectedBlocks: Set[BlockHash],
+    rejectedSenders: Set[ByteString]
+) {
+  // FringeData is uniquely identified with the hash of fringe hashes
+  // - overridden hashCode is to be more performant when used in Set or Map
+  override def hashCode(): Int = fringeHash.hashCode()
+}
+
+object FringeData {
+
+  /**
+    * Creates a hash from sorted fringe (block hashes).
+    *
+    * Used like a primary key in fringe data store.
+    */
+  def fringeHash(fringe: Set[BlockHash]) = {
+    val ordering: Ordering[ByteString] =
```

Comment:
It would be better for implicits to be separated from syntax definition so that import can be used separately.

I'll remove ordering here and use from syntax.